pegasus/routes/hourofcode_routes.rb

```diff
@@ -9,5 +9,12 @@
   only_for 'hourofcode.com'
   dont_cache unless rack_env == :production
   env['PATH_INFO'] = hoc_canonicalized_i18n_path(uri, request.query_string) unless resolve_static('public', uri)
+  # hoc_canonicalized_i18n_path has a side effect of setting some important
+  # instance variables like country, language, etc., so we need to re-assign
+  # them to actionview here.
+  #
+  # These values should ideally be directly inferred from request rather than
+  # managed as volatile instance variables.
+  update_actionview_assigns
```

Comment:
This is a step in the right direction in any case, since it means that this is the rare case where this happens (and we can now be confident that random instance variables aren't being required in other page templates) :+1: